exercises/exercise-2.md

```diff
@@ -1,5 +1,38 @@
 ---
 layout: guide
+banner: images/ex2-banner.jpg
 ---
 
 # Exercise 2
+
+In exercise 2, we are going to make a program that is capable of receiving chat messages over Bluetooth, and printing those messages to a terminal on your laptop using the USB connection.
+
+1. **Create a new project on the [online compiler](https://developer.mbed.org/compiler) based on the [Location Puck](todo) example.**
+Remember to set a unique ID in the puck `init` function, as this will make your life much easier later on.
+You can find a guide on how to create new projects in the guides section if you get stuck.
+
+1. **Create a GATT attribute for receiving chat messages using the Puck library.**
+In order to be able to receive information over Bluetooth, we need to make a writable GATT attribute.
+Make one that is 20 bytes long (the maximum length allowed for attributes).
+When you have made an attribute, compiled, flashed and run your code on the development board, you should be able to see the attribute available from your smartphone using a BLE inspector app such as [B-BLE]() or [LightBlue]().
+
+1. **Add a `onDataWritten` handler to your program.**
+Now that we are able to write data to the puck over Bluetooth using our cell phones, we should find something useful to do with this data.
+One cool thing to do is to print it out to a terminal on your laptop, so let's do that in the next step.
+
+1. **Install a terminal on your machine if you don't already have one.**
+For Windows, you can use [Termite](http://www.compuphase.com/software/termite-3.1.exe).
+For Linux, you can use [Minicom](https://help.ubuntu.com/community/Minicom).
+For OS X, you can use [CoolTerm](http://freeware.the-meiers.org/CoolTermMac.zip).
+If you are on Windows, you will also need to [install a serial driver](http://developer.mbed.org/handbook/Windows-serial-configuration).
```

Comment:
We could provide a link here to the more extensive guide on setting up serial debugging, for people to look at if they get stuck